YJIT: Add --yjit-mem-size option (#11810)
ruby/ruby: doc/yjit/yjit.md

YJIT supports all command-line options supported by upstream CRuby, but also adds a few YJIT-specific options:

- `--yjit`: enable YJIT (disabled by default)

> - `--yjit-mem-size=N`: soft limit of all memory bytes for YJIT in MiB (default: 128). Tries to limit `code_region_size + yjit_alloc_size`

Reviewer: ```suggestion
- `--yjit-mem-size=N`: soft limit on total YJIT memory consumption in MiB (default: 128). Tries to limit `code_region_size + yjit_alloc_size`
```

In the original, the "bytes" part read as a bit redundant with MiB to me.

Reviewer 2: Maybe something short like "soft limit on YJIT memory usage in MiB"
